lib/ukalloc/alloc.c

```diff
@@ -76,7 +83,11 @@ struct metadata_ifpages {
  * compile time assertion will abort the compilation and this value will have
  * to be increased.
  */
-#define METADATA_IFPAGES_SIZE_POW2 16
+#ifdef CONFIG_ARM64_FEAT_MTE
+#define METADATA_IFPAGES_SIZE_POW2 32
+#else
+#define METADATA_IFPAGES_SIZE_POW2 24
+#endif /* CONFIG_ARM64_FEAT_MTE */
```

Comment:
This was 16 previously, why is it 24 now?

Reply:
Looks like by accident, as it's definitely incorrect (24 is not a power of 2). Reverted.